enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/SdkIframeEmbedSetupProvider.tsx

```diff
@@ -43,11 +59,31 @@ export const SdkIframeEmbedSetupProvider = ({
     [settings],
   );
 
-  const updateSettings = (nextSettings: Partial<SdkIframeEmbedSettings>) =>
-    setSettings({
-      ...settings,
-      ...nextSettings,
-    } as SdkIframeEmbedSettings);
+  const updateSettings = useCallback(
+    (nextSettings: Partial<SdkIframeEmbedSettings>) =>
+      setSettings({
+        ...settings,
+        ...nextSettings,
+      } as SdkIframeEmbedSettings),
+    [settings],
+  );
+
+  useEffect(() => {
+    if (!isEmbedSettingsLoaded && !isRecentsLoading) {
+      const defaultSettings = getDefaultSdkIframeEmbedSettings(
+        "dashboard",
+        recentDashboards[0]?.id ?? EMBED_FALLBACK_DASHBOARD_ID,
+      );
+
+      updateSettings(defaultSettings);
+      setEmbedSettingsLoaded(true);
+    }
```

Comment:
Once the recents (activity logs) are finished loading, we set the default dashboard to the most recently visited dashboards. We only do this once, on page load.